Docs: delete from high availability docs references to removed configurations related to session storage

danieldspx opened the pull request:
**What this PR does / why we need it**:
Updates the documentation
**Which issue(s) this PR fixes**:

Fixes #33822

**Special notes for your reviewer**:

The issue describes the problem well. So after some time I fount that the docs are kind of outdated. You updated (b3ac63d) it saying that

> **Note:** You don't need to configure session storage, because the database will be used by default.
> If you want to offload the login session data from the database, then you can configure [remote_cache]({{< relref "../administration/configuration.md" >}}#remote-cache).

But  the `Sticky sessions` was not removed from the docs since it is not available anymore (as you will see below, this should have been removed way before this commit). The reason is the following (445b427):

> ### Auth and session token improvements
> The previous session storage implementation in Grafana was causing problems in larger HA setups due to too many write requests to the database. The remember me token also have several security issues which is why we decided to rewrite auth middleware in Grafana and remove the session storage since most operations using the session storage could be rewritten to use cookies or data already made available earlier in the request. 
If you are using `Auth proxy` for authentication the session storage will still be used but our goal is to remove this ASAP as well.
> This release will force all users to log in again since their previous token is not valid anymore.

This docs update (de2f3db) confirms:

> Grafana 6.0 removes the need of configuring and setup of additional storage for [user sessions](/tutorials/ha_setup/#user-sessions). This should make it easier to deploy and operate Grafana in a
high availability setup and/or if you're using a stateless user session storage like Redis, Memcache, Postgres or MySQL.

This commit (db584b3d) forgot to remove the discussed details from the docs. Just to really make sure we dont need [session] anymore:

> ## [session]
>**Removed starting from Grafana v6.2. Please use [remote_cache](#remote-cache) option instead.**

I think that 7899651 is the commit that missed this detail. Therefore this PR fix the doc that and closes #33822

danieldspx committed:
docs: delete from high availability docs references to removed configurations related to session storage

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=33827) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=33827) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=33827) it.</sub>

danieldspx committed:
docs: remove session storage mention and focus on the auth token implementation

danieldspx commented:
This last update was motivated by this 51c99fc:

> ### Session storage is no longer used
> In 6.2 we completely removed the backend session storage since we replaced the previous login session implementation with an auth token.

So I removed completely the metions to session storage.

achatterjee-grafana reviewed (approved):
LGTM based on https://github.com/grafana/grafana/issues/33822.
